include/orb/Orbit.h

```diff
@@ -388,5 +393,261 @@ class Orbit {
+    int _numgravcallsleft{0};
+    /** Final gps time to propagate to (ns).*/
+    uint64_t _targetgpstime{0};
+    /** The earth's angular rate in ecef frame used for the multi cycle propagation (rad/s). */
+    lin::Vector3d _earth_rate_ecef;
+    /** The maximum size of a 7 grav call higher order step (ns). */
+    GNC_TRACKED_CONSTANT(static const int64_t,maxlongtimestep,100'000'000'000LL);
+    /** The maximum size of a 1 grav call step (ns). */
+    GNC_TRACKED_CONSTANT(static const int64_t,maxshorttimestep,200'000'000LL);
+
+
+    /**
+     * Put the Orbit in propagating mode or change the end time of a propagating Orbit.
+     * Future calls to finishpropagating() or onegravcall() numgravcallsleft() times will finish the propagating.
+     * Don't do anything if the orbit is invalid.
+     *
+     * Schedual greedily by taking large steps first then smaller steps if needed.
+     * If the Orbit is inside a higher order step, don't interrupt it, just schedual onto the end.
+     *
+     * grav calls: 0
+     * @param[in] end_gps_time_ns: Time to propagate to (ns).
+     * @param[in] earth_rate_ecef: The earth's angular rate in ecef frame ignored if already propagating(rad/s).
+     */
+    void startpropagating(const uint64_t& end_gps_time_ns, const lin::Vector3d& earth_rate_ecef){
+        if (!valid()){
+            return;
+        }
+        _targetgpstime= end_gps_time_ns;
+        //now get _numgravcallsleft
+        int64_t deltatime= _targetgpstime-_ns_gps_time;
+        int64_t absdeltatime= std::abs(deltatime);
+        if((_numgravcallsleft==0) && (deltatime!=0)){
+            //starting save earth rate
+            _earth_rate_ecef= earth_rate_ecef;
+        }
+        //dont interrupt a higher order step.
+        _numgravcallsleft= _longstep?(7-_longstep):0;
+        if (absdeltatime>maxshorttimestep*6){
+            //how many long steps
+            int n= absdeltatime/maxlongtimestep;//full long steps
+            _numgravcallsleft+=n*7;
+            absdeltatime-=n*maxlongtimestep;
+        }
+        if (absdeltatime==0){
+            return;
+        }
+        int shortstepsleft=(absdeltatime-1)/maxshorttimestep+1;
+        _numgravcallsleft+=std::min(shortstepsleft,7);
+    }
+
+    /**
+     * Return the number of grav calls needed to finish propagating.
+     * If not propagating Return 0.
+     *
+     * grav calls: 0
+     */
+    int numgravcallsleft() const{
+        return _numgravcallsleft;
+    }
+
+    //attributes for relative orbit
+    /** relative time (s).*/
+    double _t{0};
+    /** higher order step total dt (s).*/
+    double _currentdt{0};
+    /** mu/(a*a*a) where a is the semimajor axis of the referance orbit (MKS units). */
+    double _mu_a3;
+    /** x axis of reference orbit (m).*/
+    lin::Vector3d _x;
+    /** y axis of reference orbit (m).*/
+    lin::Vector3d _y;
+    /** reference orbit rate (rad/s).*/
+    lin::Vector3d _omega;
```

Comment:
This x and y axis of the reference orbit is describing a "close" circular orbit with identical plane and semi-major axis? `_x` stores the positional information and `_y` is set along the velocity vector but still has an identical magnitude to `_x`?

Reply:
Yeah, kind of. Right now the equation is 
```
double theta= _t*lin::norm(_omega);
double costheta= std::cos(theta);
double sintheta= std::sin(theta);
lin::Vector3d orb_r= _x*costheta+_y*sintheta;
```
But there is a lot of redundant info here.

Reply:
Ahh okay that makes more sense! Thanks.